add api of publist/unPublish product.

iamcxa opened the pull request:
- 1st pr.

iamcxa committed:
add api of publist/unPublish product.

iamcxa committed:
- delete
	- add api for delete product.
	- add view compoment to delete product
- publish/unpublish
	- add view compoment for index, to show attributes "isPublish".
	- add button to publish or un publish a product.

iamcxa committed:
- add data field "comment" for db Product.

iamcxa committed:
- fix missing "," symbol in the SPEC of ProductController.js.
- add "?responseType=view" suffix for publish/unpublish/delete button in jade view.

iamcxa committed:
- remove .only in spec(s).

smlsunxie commented on api/controllers/ProductController.js:
```diff
@@ -73,6 +69,81 @@ let ProductController = {
+          msg: '找不到商品！ 請確認商品ID！'
+        });
+      }
+      findProduct.destroy();
```
這邊要加上  await 不然有可能會沒有執行完成

iamcxa committed:
- change "put" to "delete" prefix for Delete API to fit restful.
- adjuct spec to fix changes above.

iamcxa committed:
Merge remote-tracking branch 'upstream/master'

iamcxa committed:
- ProductController.js
 - add await for .destroy() method.
- routes.coffee
 - change http path to fit restful spec. (remove /delete/)
- ProductController.spec
 - change path to fit changes.
